am-lib/src/main/java/uk/gov/hmcts/reform/amlib/models/ExplicitAccessMetadata.java

```diff
@@ -20,7 +20,7 @@ private ExplicitAccessMetadata(String resourceId,
                                    String resourceName,
                                    JsonPointer attribute,
                                    String securityClassification) {
-        super(resourceId, accessorId, accessType, serviceName, resourceType, resourceName, attribute,
+        super(resourceId, accessorId, accessType, serviceName, resourceType, resourceName, attribute + "%",
```

Comment:
That is wrong. That class should be reusable in other places not only for revoking access.

Other thing is that when `/amount` is revoked then `/amountInPounds` for example should remained untouched. Simple ends width is not an answer. Revoking `/amount` should revoke `/amount`, `/amount/lastUpdated` etc but not `/amountInPounds` as it is absolutely different properly.

Reply:
I have updated the sql now and removed from the class and it works. 

On your second point I question how this would be done. In other instance we assume the person will be smart in what they are asking for. If they want to delete test and everything in test they would remove test/. If we were to remove the backslash at the end for a check it would remove everything and so we would not want to do this. If they want to remove test and everything in it they would pass test/ and this works.

Reply:
In my personal opinion I feel that we should have a delete and a cascade delete method separate. I feel that if someone is calling cascade delete on an attribute it is a big deal and so I feel that going forward these should be separately implemented. Documentation will need to be strong also with this current implementation as would not want someone accidentally deleting everything when they only wanted to possibly delete one attribute and accidentally deleted it and everything contained within it.

Reply:
Requirement is simple @rebeccabrennan . When somebody deletes attribute and that attribute points to complex object then system must delete that attribute and all children of that attribute.

IMO this is reasonable behaviour. It is like cutting of branch of the tree. When you do it you do not want to remove wooden part only and leave leaves that were attached to the branch. You expect expect whole thing to go away.

Reply:
👍